core/src/test/java/io/confluent/rest/ApiHeadersTest.java

```diff
@@ -0,0 +1,166 @@
+    TestSslUtils.createTrustStore(trustStore.getAbsolutePath(), new Password(SSL_PASSWORD), certs);
+
+    Properties props = new Properties();
+    props.put(RestConfig.LISTENERS_CONFIG, httpUri + "," + httpsUri);
+    props.put(RestConfig.SSL_KEYSTORE_LOCATION_CONFIG, serverKeystore.getAbsolutePath());
+    props.put(RestConfig.SSL_KEYSTORE_PASSWORD_CONFIG, SSL_PASSWORD);
+    props.put(RestConfig.SSL_KEY_PASSWORD_CONFIG, SSL_PASSWORD);
+
+    app = new TestApplication(new TestRestConfig(props));
+    app.start();
+  }
+
+  @AfterClass
+  public static void teardown() throws Exception {
+    if (app != null) {
+      app.stop();
+    }
+  }
+
+  private static void createKeystoreWithCert(File file, String alias, Map<String, X509Certificate> certs) throws Exception {
+    KeyPair keypair = TestSslUtils.generateKeyPair("RSA");
+    CertificateBuilder certificateBuilder = new CertificateBuilder(30, "SHA1withRSA");
+    X509Certificate cCert = certificateBuilder.sanDnsName("localhost")
+        .generate("CN=mymachine.local, O=A client", keypair);
+    TestSslUtils.createKeyStore(file.getPath(), new Password(SSL_PASSWORD), alias, keypair.getPrivate(), cCert);
+    certs.put(alias, cCert);
+  }
+
+  @Test
+  public void testHttpDoesNotReturnJettyServerVersionHeader() throws Exception {
+
+    HttpGet httpget = new HttpGet(httpUri +  "/test/endpoint");
+
+    try ( CloseableHttpClient httpclient = HttpClients.createDefault();
+        CloseableHttpResponse response = httpclient.execute(httpget) ) {
+
+      int statusCode = response.getStatusLine().getStatusCode();
+      Assert.assertEquals(200, statusCode);
+
+      Header jettyServerVersionHeader = response.getFirstHeader( "Server" );
+      assertNull(jettyServerVersionHeader);
+    }
+  }
+
+  @Test
+  public void testHttpsDoesNotReturnJettyServerVersionHeader() throws Exception {
+
+    HttpGet httpget = new HttpGet(httpsUri +  "/test/endpoint");
+
+    // trust all self-signed certs.
+    SSLContextBuilder sslContextBuilder = SSLContexts.custom()
+        .loadTrustMaterial(new TrustSelfSignedStrategy());
+
+    // add the client keystore if it's configured.
+      sslContextBuilder.loadKeyMaterial(new File(clientKeystoreLocation),
+          SSL_PASSWORD.toCharArray(), SSL_PASSWORD.toCharArray());
+    SSLContext sslContext = sslContextBuilder.build();
+
+    SSLConnectionSocketFactory sslSf = new SSLConnectionSocketFactory(sslContext, new String[]{"TLSv1.2"},
+        null, SSLConnectionSocketFactory.getDefaultHostnameVerifier());
+
+    try ( CloseableHttpClient httpclient = HttpClients.custom().setSSLSocketFactory(sslSf).build();
+        CloseableHttpResponse response = httpclient.execute(httpget) ) {
+
+      int statusCode = response.getStatusLine().getStatusCode();
+      Assert.assertEquals(200, statusCode);
+
+      Header jettyServerVersionHeader = response.getFirstHeader( "Server" );
+      assertNull(jettyServerVersionHeader);
```

Comment:
nit: use Hamcrest's `assertThat` for all new code.

```suggestion
      assertThat(jettyServerVersionHeader, is(nullValue()));
```

Reply:
Done